pkg/render/intrusion_detection.go

```diff
@@ -155,11 +190,33 @@ func (c *intrusionDetectionComponent) Objects() ([]client.Object, []client.Objec
 			c.intrusionDetectionPSPClusterRoleBinding())
 	}
 
+	if c.dpiDaemonSet {
+		c.intrusionDetectionComponentResources()
+		objs = append(objs, CreateNamespace(DeepPacketInspectionNamespace, c.installation.KubernetesProvider))
+		objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(DeepPacketInspectionNamespace, c.nodeTLSSecret)...)...)
+		objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(DeepPacketInspectionNamespace, c.typhaTLSSecret)...)...)
+		objs = append(objs, configmap.ToRuntimeObjects(configmap.CopyToNamespace(DeepPacketInspectionNamespace, c.typhaCAConfigMap)...)...)
+		objs = append(objs,
+			c.dpiServiceAccount(),
+			c.dpiRole(),
+			c.dpiRoleBinding(),
+			c.dpiDaemonset(),
+		)
+	} else {
+		toDelete = append(toDelete, c.dpiNamespace())
+	}
+
 	if c.hasNoLicense {
 		return nil, objs
 	}
 
-	return objs, nil
+	// If IDS resource needs to be updated with default resourceRequirements,
+	// append it after license check is done to avoid deleting IDS CR.
+	if c.dpiDaemonSet {
+		objs = append(objs, c.ids)
```

Comment:
Could we have some built-in defaults in the API instead of trying to populate them in our render ?

Reply:
Tried setting defaults using kubebuilder and json tags on the crd, it doesn't reflect on the intrusiondetections resource (I didn't delete and recreate the resource after applying updated crd), I'll leave this be.